### PR TITLE
Substitute nested `ifelse` with `data.table::fcase()`

### LINKED ISSUE #379: Use of nested `ifelse`
The current code base has multiple instances of nested `ifelse` usage. These would be easier to manage if switched over to using `data.table::fcase()`.

### PR DESCRIPTION
**Pull request by jamesmbaazam**
This PR replaces all occurrences of nested `ifelse` with `data.table::fcase()` and fixes #379. Each commit mentions the function that was changed, making it easier to roll back to in case of any consequences of the change.

### CODE REVIEW

File: R/dist.R
Comment (seabbs) [inline R/dist.R:201]: These are all nice changes @jamesmbaazam but if possible can you address in separate issue/PR combos vs here as otherwise this PR may be hard to review. ty ❤️

Comment (jamesmbaazam) [reply R/dist.R:201]: Agreed. I'll do that.

**Comment by seabbs** [comment]
Are you waiting for review on this @jamesmbaazam? Before I can review it would ideally be passing CI checks or there would be some detail on why it isn't and why that is okay

**Comment by jamesmbaazam** [comment]
No, I'm not waiting. I am about to try and fix the checks.

**Comment by jamesmbaazam** [comment]
No success so far! The problem seems to come from somewhere within `epinow()` but not exactly where. I will keep looking!

File: R/dist.R
Comment (seabbs) [inline R/dist.R:201]: just need to redoc to get rid of the warning here.

Comment (jamesmbaazam) [reply R/dist.R]: I've reverted this change. I'll submit it as a separate PR on fixing defaults.

File: R/create.R
Comment (sbfnk) [inline R/create.R:596]: Stan expects `rho` to be an array so I think you'll need

```suggestion
      out$rho <- array(data.table::fcase(
        out$rho > data$ls_max, data$ls_max - 0.001,
        out$rho < data$ls_min, data$ls_min + 0.001,
        default = out$rho
        ))
```

If doing this you can remove the `array` a few lines above and just do `out$rho <- rlnorm`...

Comment (jamesmbaazam) [reply R/create.R]: Nice catch. Thanks, Seb. My knowledge of stan is too limited for me to have caught that.

File: R/estimate_infections.R
Comment (jamesmbaazam) [reply R/estimate_infections.R:705]: I interpret the final return of the original `fifelse` statement as "if it's neither within the horizon or the horizon and shift, then make it 'estimate'". Am I wrong?

Comment (sbfnk) [reply R/estimate_infections.R:705]: Yes, `fifelse` has an `na` argument that determines what happens if the test yields NA. 
https://rdatatable.gitlab.io/data.table/reference/fifelse.html

Comment (sbfnk) [reply R/estimate_infections.R:705]: Yes, `fifelse` has an `na` argument that determines what happens if the test yields NA. 
https://rdatatable.gitlab.io/data.table/reference/fifelse.html

Comment (sbfnk) [reply R/estimate_infections.R:705]: Yes, `fifelse` has an `na` argument that determines what happens if the test yields NA. 
https://rdatatable.gitlab.io/data.table/reference/fifelse.html

Comment (jamesmbaazam) [reply R/estimate_infections.R:705]: Thanks for pointing that out. I missed that.

Comment (sbfnk) [reply R/estimate_infections.R:705]: Sorry for message spam (from shabby internet connection)!

Comment (jamesmbaazam) [reply R/estimate_infections.R:705]: I figure :)

File: R/create.R
Comment (seabbs) [reply R/create.R]: Sorry I thought I commented this but it was caught up as pending. Good spot. Generally if tests are failing checking if anything is different about the parts of the code base can be a good debug if you don't really get how they are later used. 

> this seems to be an issue when passed to stan (the dimension warning). It might be as simple as wrapping this in an array call.

**Comment by seabbs** [comment]
Checks look like they are passing - nice one. Ready for review?

**Comment by jamesmbaazam** [comment]
Yes, you can review it now. Also, who resolves the conversations? The maintainer or the one who opened the PR?

**Comment by seabbs** [comment]
> Yes, you can review it now. Also, who resolves the conversations? The maintainer or the one who opened the PR?

Can I? What a treat 😆 (joking)

Typically they are resolved by the person who deals with them (i.e. you in this instance) and then when you request a review it can be helpful to point out how you have done this (or do it in the comment thread).

**Comment by jamesmbaazam** [comment]
Thanks. I saw you resolved some of our earlier conversations, so I was wondering if you preferred to resolve them.

File: R/create.R
Comment (seabbs) [inline R/create.R:289]: We can drop these linting gates now as this should pass linting checks.

File: R/create.R
Comment (seabbs) [inline R/create.R:354]: are these indented correctly? It looks like it has changed but perhaps that is just an artefact?

File: R/create.R
Comment (seabbs) [inline R/create.R:362]: remove linting gates

File: R/dist.R
Comment (seabbs) [inline R/dist.R:208]: this seems like it isn't a required or desired change?

File: R/estimate_infections.R
Comment (seabbs) [inline R/estimate_infections.R:705]: Is NA_character the right choice for a date (I don't know). Can Na handling not be set directly using `fcase`?

File: R/estimate_infections.R
Comment (seabbs) [inline R/estimate_infections.R:707]: is this formatted correctly?

File: R/estimate_infections.R
Comment (seabbs) [inline R/estimate_infections.R:708]: Drop lint gates

File: R/stanmodels.R
Comment (seabbs) [inline R/stanmodels.R:20]: don't edit stanmodels (its automated)

File: R/summarise.R
Comment (seabbs) [inline R/summarise.R:674]: remove no lint gates if ifelse linting is fixed.

**Review by seabbs — CHANGES_REQUESTED** [review]
Just a few details require attention. Also can you add yourself to the description as a contributor. Thanks!

**Review by seabbs — CHANGES_REQUESTED** [review]
Just a few details require attention. Also can you add yourself to the description as a contributor. Thanks!

**Comment by seabbs** [comment]
We also need a dev version increment and a news item for this PR.

File: R/estimate_infections.R
Comment (sbfnk) [reply R/estimate_infections.R:705]: It's a string (vs. "forecast", "estimate", etc.). I don't think `fcase` has any `NA` handling sadly (unlike `fifelse`). Perhaps to avoid confunsion perhaps the explicit handling isn't so bad though.

File: R/create.R
Comment (jamesmbaazam) [reply R/create.R]: This has been fixed in 7d30e3b4.

File: R/create.R
Comment (jamesmbaazam) [reply R/create.R]: This has been fixed in 7d30e3b4.

File: R/estimate_infections.R
Comment (jamesmbaazam) [reply R/estimate_infections.R:707]: This has been fixed in 7d30e3b4.

File: R/dist.R
Comment (jamesmbaazam) [reply R/dist.R:208]: I reverted this change in 7b63d9bf79b1c30ae15df7780e2a013b33223927. Are you reviewing sequentially by commit?

File: R/create.R
Comment (jamesmbaazam) [reply R/create.R:352]: Looks alright on my end.

File: R/create.R
Comment (seabbs) [inline R/create.R:360]: random space in a list?

File: R/create.R
Comment (seabbs) [reply R/create.R:352]: why are they showing up as changes? Were they not correctly indented before?

File: R/create.R
Comment (jamesmbaazam) [reply R/create.R]: Weird, isn't it? Fixed in bf9459c363c6804400445599532ccb4fa3174296.

**Comment by jamesmbaazam** [comment]
I've incremented the dev version and added a news item here 276af42c192613a78ea60e995716a9443c5d8c69.

File: R/stanmodels.R
Comment (jamesmbaazam) [reply R/stanmodels.R:20]: Yes, I do, I just confirmed. I was surprised that file kept getting edited.

File: R/create.R
Comment (jamesmbaazam) [reply R/create.R:352]: I styled each part where I made a change, so that's probably why.

File: DESCRIPTION
Comment (seabbs) [inline DESCRIPTION:5]: Dev version needs to go up in increments of a 1000.

File: R/dist.R
Comment (seabbs) [inline R/dist.R:209]: This is unrelated to this or. No changes shoul show up here.

File: R/estimate_infections.R
Comment (seabbs) [inline R/estimate_infections.R:707]: This doesn't look like the right formatting

File: DESCRIPTION
Comment (jamesmbaazam) [reply DESCRIPTION:5]: This has been fixed in b853c987.

File: NEWS.md
Comment (jamesmbaazam) [reply NEWS.md:3]: This has been fixed in 37786f08.

File: NEWS.md
Comment (jamesmbaazam) [reply NEWS.md:15]: This has been addressed in b286eba3

File: R/create.R
Comment (jamesmbaazam) [reply R/create.R:352]: I really don't know why it's this way. It's not indented on my machine.

Comment (jamesmbaazam) [reply R/create.R:352]: I really don't know why it's this way. It's not indented on my machine.

Comment (jamesmbaazam) [reply R/create.R]: I have tried addressing it here e8836198bdd9a0882cc7a8d7a24065c5141410cb. Do let me know if it's fixed.

Comment (jamesmbaazam) [reply R/create.R]: I have tried addressing it here e8836198bdd9a0882cc7a8d7a24065c5141410cb. Do let me know if it's fixed.

File: R/estimate_infections.R
Comment (jamesmbaazam) [reply R/estimate_infections.R]: I have tried addressing it here e8836198bdd9a0882cc7a8d7a24065c5141410cb. Do let me know if it's fixed.

File: R/create.R
Comment (seabbs) [inline R/create.R:329]: Space not matching package style

**Review by seabbs — APPROVED** [review]
LGTM to me - thanks James. 

Maybe worth setting up a call between you, seb, and me to debrief how you felt about this and things we could do better in the review process. 

Just waiting for checks to pass and then will merge.

**Review by seabbs — APPROVED** [review]
LGTM to me - thanks James. 

Maybe worth setting up a call between you, seb, and me to debrief how you felt about this and things we could do better in the review process. 

Just waiting for checks to pass and then will merge.

**Comment by jamesmbaazam** [comment]
Thanks. Agreed!

**Pull request merged**
